datumaro/components/explorer.py

```diff
@@ -43,22 +51,29 @@ def __init__(
         """
         self._model = ExplorerLauncher(model_name="clip_visual_ViT-B_32")
         self._text_model = ExplorerLauncher(model_name="clip_text_ViT-B_32")
-        inference = dataset.run_model(self._model, append_annotation=True)
         self._topk = topk
-
         database_keys = []
         item_list = []
 
-        for item in inference:
-            for annotation in item.annotations:
-                if isinstance(annotation, HashKey):
-                    try:
-                        hash_key = annotation.hash_key[0]
-                        hash_key = np.unpackbits(hash_key, axis=-1)
-                        database_keys.append(hash_key)
-                        item_list.append(item)
-                    except Exception:
-                        hash_key = None
+        datasets = datasets[0]
+        if isinstance(datasets, IDataset):
+            datasets = [datasets]
+
```

Comment:
```suggestion
```

Reply:
This part is needed to use first one of `datasets`. `datasets` is tuple and we need to use the first one.

Reply:
I don't understand what you mean. You give a sequence of `Dataset` to `Explore`.

https://github.com/openvinotoolkit/datumaro/blob/1b8c8f4399a00cc97720ce4d3b8abe103699ecac/datumaro/cli/commands/explore.py#L109-L119

But, why do we have to just use the first one `Dataset` of the sequence? Didn't you want to explore all datasets present in the sequence? If not, isn't it correct to only accept a single dataset as the input?

Reply:
The possible input for `datasets` are `Union[List[IDataset], IDataset]`, so the first one of datasets as input in now is like `([dataset1, dataset2],)` or `(dataset1,)`. To avoid this, I need to convert `source_datasets` in line 112 to tuple which you attached above. If it is better, I will change it to use tuple instead of list.

Reply:
I think that you can just change L119 to `explorer = Explorer(*source_datasets)`.